x-pack/plugins/monitoring/server/lib/cluster/get_index_patterns.ts

```diff
@@ -123,7 +123,9 @@ export function getDsIndexPattern({
   config,
   ccs,
 }: CommonIndexPatternArgs & { type?: string }): string {
-  const datasetsPattern = `${moduleType ?? '*'}.${dataset ?? '*'}`;
+  const datasetsPattern = `${moduleType ?? '*'}.${
+    type === DS_INDEX_PATTERN_METRICS ? 'stack_monitoring.' : ''
```

Comment:
Added this condition when resolving conflicts and looks like this function is doing too much. We could split it in `getMetricsIndexPatterns` and `getLogsIndexPatterns`. the split itself is reasonable and makes more sense with the dataset branching now

Reply:
thanks @klacabane. I'll try to improve this.